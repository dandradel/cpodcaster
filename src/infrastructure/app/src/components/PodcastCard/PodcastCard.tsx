/* eslint-disable @typescript-eslint/no-explicit-any */
import React, { ReactElement } from 'react'
import { Box, Divider, Typography } from '@mui/material'
import './PodcastCard.styles.css'
import {
  podcastCardBoxStyles,
  podcastImageBoxStyles,
  podcastCardTextBoxStyles,
  podcastDescriptionTextStyles,
  podcastTextStyles,
} from './utils'
import { useNavigate } from 'react-router-dom'

export interface PodcastCardProps {
  podcastImage?: string
  podcastName?: string
  podcastAuthor?: string
  podcastDescription?: string
  podcastId?: string | number
}

export interface PodcastCardArgs {
  podcastCard: PodcastCardProps
}

function PodcastCard({ podcastCard }: PodcastCardArgs): ReactElement {
  const navigate = useNavigate()

  const {
    podcastImage = '',
    podcastName = '',
    podcastAuthor = '',
    podcastDescription = '',
    podcastId = '',
  } = podcastCard

  const handlePodcastCardRedirectionClick = (): void => {
    navigate(`/podcast/${podcastId}`)
  }

  return (
    <Box sx={podcastCardBoxStyles} data-testid='podcast-wrapper-testid'>
      <Box sx={podcastImageBoxStyles}>
        <img
          className='podcastCard__image'
          src={podcastImage}
          onClick={handlePodcastCardRedirectionClick}
          data-testid='podcast-image-testid'
        />
      </Box>
      <Divider variant='middle' />
      <Box sx={podcastCardTextBoxStyles}>
        <Typography
          variant='subtitle1'
          fontWeight='bold'
          gutterBottom
          data-testid='podcast-name-testid'
          onClick={handlePodcastCardRedirectionClick}
          sx={podcastTextStyles}
        >
          {podcastName}
        </Typography>
        <Typography
          variant='subtitle2'
          fontStyle='italic'
          gutterBottom
          data-testid='podcast-author-testid'
          onClick={handlePodcastCardRedirectionClick}
          sx={podcastTextStyles}
        >
          by {podcastAuthor}
        </Typography>
      </Box>
      <Divider variant='middle' />
      <Box sx={podcastDescriptionTextStyles}>
        <Typography
          variant='subtitle2'
          fontWeight='bold'
          gutterBottom
          data-testid='podcast-description-testid'
        >
          Description:
        </Typography>
        <Typography
          variant='subtitle2'
          fontStyle='italic'
          gutterBottom
          data-testid='podcast-description1-testid'
        >
          {podcastDescription}
        </Typography>
      </Box>
    </Box>
  )
}

export default PodcastCard
