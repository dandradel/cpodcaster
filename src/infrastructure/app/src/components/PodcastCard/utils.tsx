import { SxProps } from '@mui/material'

export const podcastCardBoxStyles: SxProps = {
  display: 'flex',
  flexDirection: 'column',
  flexBasis: '100%',
  flex: 0.5,
  maxHeight: 500,
  maxWidth: 300,
  bgcolor: 'white',
  boxShadow: 3,
  borderRadius: 1,
}
export const podcastImageBoxStyles: SxProps = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  my: 2,
}
export const podcastCardTextBoxStyles: SxProps = {
  display: 'block',
  alignItems: 'left',
  justifyContent: 'left',
  p: 2,
  m: 1,
}
export const podcastDescriptionTextStyles: SxProps = {
  display: 'block',
  alignItems: 'left',
  justifyContent: 'left',
  p: 2,
}
export const podcastTextStyles: SxProps = { cursor: 'pointer' }
