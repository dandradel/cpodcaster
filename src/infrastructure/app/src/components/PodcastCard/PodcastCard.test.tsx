import React from 'react'
import '@testing-library/jest-dom'
import { cleanup, renderWithProviders, screen } from '../../shared/test/test-utils'
import PodcastCard, { PodcastCardProps } from './PodcastCard'
import podcastMock from '../../shared/__mocks__/podcasts.json'

describe('PodcastCard', () => {
  afterEach(cleanup)
  const podcastCardProps: PodcastCardProps = {
    podcastImage: podcastMock[0].image,
    podcastName: podcastMock[0].name,
    podcastAuthor: podcastMock[0].author,
    podcastDescription: '',
  }

  test('renders PodcastCard', () => {
    const container = renderWithProviders(<PodcastCard podcastCard={podcastCardProps} />)
    expect(container).toMatchSnapshot()
  })
  test('renders PodcastCard with name', () => {
    renderWithProviders(<PodcastCard podcastCard={podcastCardProps} />)
    const podcastsNames = screen.getByTestId('podcast-name-testid')
    expect(podcastsNames).toHaveTextContent(podcastCardProps.podcastName || '')
  })
  test('renders PodcastCard with author', () => {
    renderWithProviders(<PodcastCard podcastCard={podcastCardProps} />)
    const podcastsAuthor = screen.getByTestId('podcast-author-testid')
    expect(podcastsAuthor).toHaveTextContent(`by ${podcastCardProps.podcastAuthor}`)
  })
  test('renders PodcastCard with description', () => {
    renderWithProviders(<PodcastCard podcastCard={podcastCardProps} />)
    const podcastsDescription = screen.getByTestId('podcast-description-testid')
    expect(podcastsDescription).toHaveTextContent('Description:')
  })
  test('renders PodcastCard with description1', () => {
    renderWithProviders(<PodcastCard podcastCard={podcastCardProps} />)
    const podcastsDescription1 = screen.getByTestId('podcast-description1-testid')
    expect(podcastsDescription1).toHaveTextContent(podcastCardProps.podcastDescription || '')
  })
})
