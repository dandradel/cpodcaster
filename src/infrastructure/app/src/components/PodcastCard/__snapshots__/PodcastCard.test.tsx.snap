// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastCard renders PodcastCard 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-6m2utx"
        data-testid="podcast-wrapper-testid"
      >
        <div
          class="MuiBox-root css-1ia54b9"
        >
          <img
            class="podcastCard__image"
            data-testid="podcast-image-testid"
            src="https://picsum.photos/200/300"
          />
        </div>
        <hr
          class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-1d8isoc"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-165m6tq-MuiTypography-root"
            data-testid="podcast-name-testid"
          >
            Podcast 1
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-t9vmtp-MuiTypography-root"
            data-testid="podcast-author-testid"
          >
            by 
            Author 1
          </h6>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-tcr4li"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-16yea3k-MuiTypography-root"
            data-testid="podcast-description-testid"
          >
            Description:
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-4qfdts-MuiTypography-root"
            data-testid="podcast-description1-testid"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-6m2utx"
      data-testid="podcast-wrapper-testid"
    >
      <div
        class="MuiBox-root css-1ia54b9"
      >
        <img
          class="podcastCard__image"
          data-testid="podcast-image-testid"
          src="https://picsum.photos/200/300"
        />
      </div>
      <hr
        class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1d8isoc"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-165m6tq-MuiTypography-root"
          data-testid="podcast-name-testid"
        >
          Podcast 1
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-t9vmtp-MuiTypography-root"
          data-testid="podcast-author-testid"
        >
          by 
          Author 1
        </h6>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-tcr4li"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-16yea3k-MuiTypography-root"
          data-testid="podcast-description-testid"
        >
          Description:
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-4qfdts-MuiTypography-root"
          data-testid="podcast-description1-testid"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
